File: app/models/Item.php
<?php
class Item extends Model {

  function __construct($id='')
  {
    parent::__construct('Id','Item'); //primary key = uid; tablename = users
    $this->rs['Id'] = '';
    $this->rs['Name'] = '';
    $this->rs['Description'] = '';
    $this->rs['Price'] = '';
    $this->rs['Sku'] = '';
    $this->rs['CategoryId'] = '';
    $this->rs['CollectionId'] = '';
    $this->rs['AvailableInventory'] = '';
    $this->rs['CoverPhotoId'] = '';
    $this->rs['Date'] = '';
    if ($id)
      $this->retrieve($id);
  }

  function create()
  {
    $this->rs['Date']=date('Y-m-d H:i:s');
    return parent::create();
  }

  function delete()
  {
  	$photo = new Picture();
  	$photoarr = $photo->retrieve_many('ItemId = ?', $this->rs['Id']);

  	for($i = 0; $i < sizeof($photoarr); $i++)
  	{
  		//delete image file
  		unlink($photoarr[$i]->rs['Data']);
  	}

  	return parent::delete();
  }
}

File: Malikah.Api/Data/Entities/Item.cs
using System;
using System.Collections.Generic;

namespace Malikah.Api.Data.Entities
{
    public partial class Item
    {
        public Item()
        {
            Itemvariation = new HashSet<Itemvariation>();
            Picture = new HashSet<Picture>();
            Review = new HashSet<Review>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Price { get; set; }
        public string Sku { get; set; }
        public int CollectionId { get; set; }
        public int? CategoryId { get; set; }
        public int AvailableInventory { get; set; }
        public int CoverPhotoId { get; set; }

        public Collection Collection { get; set; }
        public ICollection<Itemvariation> Itemvariation { get; set; }
        public ICollection<Picture> Picture { get; set; }
        public ICollection<Review> Review { get; set; }
    }
}
